src/plugins/data/common/search/aggs/metrics/lib/get_response_agg_config_class.ts

```diff
@@ -46,7 +46,7 @@ export const create = (parentAgg: IMetricAggConfig, props: Partial<IMetricAggCon
     const subId = String(key);
 
     if (subId.indexOf('.') > -1) {
-      id = parentId + "['" + subId.replace(/'/g, "\\'") + "']";
+      id = parentId + "['" + subId.replace(/[\\']/g, '\\$&') + "']"; // $& means the whole matched string
```

Comment:
I would be more confortable to extract this logic out into an independent function and provide some unit tests to avoid regressions in case somebody changes it in the future.

Something simple as:
```ts
id = escapeChars(parentId, subId);
```

and the new function:
```ts
export function escapedChars(parentId, subId){
  return parentId + "['" + subId.replace(/[\\']/g, '\\$&') + "']"; // $& means the whole matched string
}
```

Reply:
:+1: b4e4a15f09441ef2304b74278f6d908a91fa512e